probe/appclient/app_client.go

```diff
@@ -311,7 +311,10 @@ func (c *appClient) pipeConnection(id string, pipe xfer.Pipe) (bool, error) {
 	defer c.closeConn(id)
 
 	_, remote := pipe.Ends()
-	return false, pipe.CopyToWebsocket(remote, conn)
+	if err := pipe.CopyToWebsocket(remote, conn); err != nil && !xfer.IsExpectedWSCloseError(err) {
+		return false, err
+	}
+	return false, nil
```

Comment:
TBH not sure, its been a long time since I worked on this.  There is a separate mechanism to decide when to stop retrying (the 404 above).  Originally it was supposed to retry / backoff until this is hit.